src/types/type-declarations.ts

```diff
@@ -79,6 +79,8 @@ export interface GTag {
   (fn: 'config', opt: 'reset'): void;
 }
 
+export type DataLayer = Record<string, unknown>[];
```

Comment:
doesn't every entry need an `event`? this would also allow downstream deps to augment it if needed

```ts
type DataLayer = {
  event: string
  [key: string]: any
}
```